frontend/packages/local-storage-operator-plugin/integration-tests/tests/disks-list-page.scenario.ts

```diff
@@ -0,0 +1,91 @@
+import { $, ExpectedConditions as until, browser, Key } from 'protractor';
+import { execSync } from 'child_process';
+import { appHost } from '@console/internal-integration-tests/protractor.conf';
+import { resourceRows, isLoaded } from '@console/internal-integration-tests/views/crud.view';
+import { clickNavLink } from '@console/internal-integration-tests/views/sidenav.view';
+import {
+  page,
+  diskList,
+  clickFilterDropdown,
+  checkDiskFilter,
+  clearDiskFilter,
+} from '../views/disks-list-page.views';
+import { LOCAL_STORAGE_NAMESPACE } from '../../src/constants/index';
+import {
+  LocalVolumeDiscoveryResultKind,
+  DiskMetadata,
+  DiskStates,
+} from '../../src/components/disks-list/types';
+
+// Pre-requirement: Local volume discovery to be present
+describe('Disk List is accessible from Nodes view', () => {
+  let discoveredDevices: DiskMetadata[];
+  beforeAll(async () => {
+    const lvdrJson = JSON.parse(
+      execSync(
+        `kubectl get localvolumediscoveryresults -n ${LOCAL_STORAGE_NAMESPACE} -o json`,
+      ).toString(),
+    );
+    const lvdr: LocalVolumeDiscoveryResultKind = lvdrJson.items[0];
+    const { nodeName } = lvdr.spec;
+    discoveredDevices = lvdr.status.discoveredDevices;
+    await browser.get(`${appHost}/`);
+    await clickNavLink(['Compute', 'Nodes']);
+    await isLoaded();
+    $(`a[data-test-id="${nodeName}"]`).click();
+  });
+  it('`Disks` tab is enabled and accessible', async () => {
+    await browser.wait(until.presenceOf(page.diskTab), 5000);
+    page.diskTab.click();
+  });
+  it('List title is `Disks`', async () => {
+    await isLoaded();
+    expect(page.resourceTitles.get(1).getText()).toBe('Disks');
+  });
+  it('Headers are displayed with correct labels', () => {
+    diskList.headers.forEach(async (header) => {
+      const headers = await $(`th[data-label="${header}"]`);
+      expect(headers.isPresent()).toBeTruthy();
+    });
+  });
+  it('Rows are displayed with correct data format', async () => {
+    const rows = resourceRows;
+    expect(rows.count()).toBe(discoveredDevices.length);
+    const firstRow = rows.first().$$('td');
+    const firstDisk = discoveredDevices[0];
+    expect(firstRow.get(0).getText()).toBe(firstDisk.path);
+    expect(firstRow.get(1).getText()).toBe(firstDisk.status.state);
+    expect(firstRow.get(2).getAttribute('innerText')).toBe(firstDisk.type || '-');
+    expect(firstRow.get(3).getAttribute('innerText')).toBe(firstDisk.model || '-');
```

Comment:
`getText` also returns the visible inner text of the element. I am not sure why we would want to do it this way.

Reply:
`getAttribute` returns text for hidden elements and on certain screens few of the columns will be hidden

Reply:
since it will be running headless if `getText` works on headless mode then let us keep it as `getText`

Reply:
No, `getText` , does not works therefore we are using `getAttribute` here.

Reply:
https://github.com/openshift/console/pull/6606/files#diff-1e194e3ba92229ea049eea86f7bf2e94R47
If all the headers(columns) are visible I am confused why exactly some of the elements are missing here.

Reply:
for headers the presence of DOM node is checked by the attribute data-label=<header-name>.
We are not using protractor get() and getText() functions which work on element type.

Reply:
I have posted the screenshot of running tests in description as well.
https://github.com/openshift/console/pull/6606#issue-486324168